### DISCUSSION TITLE
Include the chain ID and the Lockup model in the ID itself

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#909] [Ideas]
- Benefits: see Fjord's feedback below
- Cons: increased gas cost

<img width="450" src="https://github.com/sablier-labs/v2-core/assets/8782666/e6978dc9-4899-49d9-9add-be939dda894b"/>

<img width="450" alt="SCR-20240425-sgul" src="https://github.com/sablier-labs/v2-core/assets/8782666/c85aa49a-d3cb-40e3-8d54-36654e83a975">

**Comment by smol-ninja** [comment]
- Why the chain ID?
- If we include Lockup model in the ID, that means we will have two token IDs: 1. ERC-721 token ID and another string formatted token ID.

**Reply by PaulRBerg** [reply]
1. To make IDs unique across chains
2. That alone is a strong enough reason to discard this proposal

**Reply by PaulRBerg** [reply]
Closing the discussion for now but we should monitor user feedback and explore potential solutions if this continues to be a problem
